internal/commands/create.go

```diff
@@ -289,23 +298,63 @@ func getRegoFilePaths(path string) ([]string, error) {
 	return regoFilePaths, nil
 }
 
-func loadRegoFiles(filePaths []string) ([]regoFile, error) {
-	var regoFiles []regoFile
-	for _, filePath := range filePaths {
-		data, err := ioutil.ReadFile(filePath)
+func loadPolicyFiles(filePaths []string) ([]regoFile, error) {
+	var policyFiles []regoFile
+	policyFilesContents, err := readFilesContents(filePaths)
+	if err != nil {
+		return nil, fmt.Errorf("reading files: %w", err)
+	}
+
+	for path, contents := range policyFilesContents {
+		module, err := ast.ParseModule(path, contents)
 		if err != nil {
-			return nil, fmt.Errorf("read policy file: %w", err)
+			return nil, fmt.Errorf("parse module: %w", err)
 		}
 
-		regoFile, err := newRegoFile(filePath, string(data))
+		if moduleHasViolationRule(module) {
```

Comment:
nit: To reduce nesting we should check `!moduleHasRuleViolation` and then `continue`